### PR TITLE
Ciac 10634 prisma cloud extend commands

### PR DESCRIPTION
**Pull request by ilaredo**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-10634](https://jira-dc.paloaltonetworks.com/browse/CIAC-10634)

## Description
Add commands to programmatically create, list, and delete IAM access keys via the Access Keys API endpoints for the management of service account keys.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 1]
```diff
@@ -2131,6 +2131,52 @@ script:
     description: Updates the remote incident with local incident changes. This method is only used for debugging purposes and will not update the current incident.
     execution: false
     name: update-remote-system
+  - arguments:
+      - description: Access key name
+        name: name
+        required: true
+      - description: Timestamp in milliseconds when access key expires. Default:0
+        name: expires-on
+      - description: Service account name
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:2140]:
```suggestion
      - description: Access key name.
        name: name
        required: true
      - description: Timestamp in milliseconds when access key expires. Default:0.
        name: expires-on
      - description: Service account name.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 2]
```diff
@@ -2833,6 +2833,139 @@ Get permission list. You must provide either "query" or "next_token".
...
 
+`prisma-cloud-access-key-create`
+
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+|-------------------| --- | --- |
+| name              | Access key name | Required | 
+| expires-on             | Timestamp in milliseconds when access key expires. Default:0 | Optional | 
+| service-account-name             | Service account name | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:2844]:
```suggestion
| name              | Access key name. | Required | 
| expires-on             | Timestamp in milliseconds when access key expires. Default:0. | Optional | 
| service-account-name             | Service account name. | Optional |
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 3]
```diff
@@ -2833,6 +2833,139 @@ Get permission list. You must provide either "query" or "next_token".
...
+
+#### Context Output
+
+| **Path**                  | **Type** | **Description**       |
+|---------------------------| --- |-----------------------|
+| PrismaCloud.AccessKeys.id  | String | Access key ID        | 
+| PrismaCloud.AccessKeys.secretKey | String | Access key secret |
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:2851]:
```suggestion
| PrismaCloud.AccessKeys.id  | String | Access key ID.        | 
| PrismaCloud.AccessKeys.secretKey | String | Access key secret. |
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 4]
```diff
@@ -2833,6 +2833,139 @@ Get permission list. You must provide either "query" or "next_token".
...
+| PrismaCloud.AccessKeys.name        | String   | Name                        |
+| PrismaCloud.AccessKeys.createdBy   | String   | Created By                  | 
+| PrismaCloud.AccessKeys.createdTs   | String   | Time access key was created |
+| PrismaCloud.AccessKeys.expiresOn   | String   | Access key expires on       |
+| PrismaCloud.AccessKeys.lastUsedTime | String   | Access key last used        | 
+| PrismaCloud.AccessKeys.status      | String   | Access key status           |
+| PrismaCloud.AccessKeys.role.id     | String   | User role id                | 
+| PrismaCloud.AccessKeys.role.name   | String   | User role name              |
+| PrismaCloud.AccessKeys.roleType    | String   | User role permission type   | 
+| PrismaCloud.AccessKeys.username    | String   | Access key user name        |
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:2892]:
```suggestion
| PrismaCloud.AccessKeys.id          | String   | Access key ID.               | 
| PrismaCloud.AccessKeys.name        | String   | The name of the access key.                        |
| PrismaCloud.AccessKeys.createdBy   | String   | The user who created the access key.                  | 
| PrismaCloud.AccessKeys.createdTs   | String   | Time access key was created. |
| PrismaCloud.AccessKeys.expiresOn   | String   | Access key expires on.       |
| PrismaCloud.AccessKeys.lastUsedTime | String   | The last time the access key was used.        | 
| PrismaCloud.AccessKeys.status      | String   | Access key status.           |
| PrismaCloud.AccessKeys.role.id     | String   | User role ID.       | 
| PrismaCloud.AccessKeys.role.name   | String   | User role name.              |
| PrismaCloud.AccessKeys.roleType    | String   | User role permission type.   | 
| PrismaCloud.AccessKeys.username    | String   | Access key user name.        |
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 5]
```diff
@@ -2833,6 +2833,139 @@ Get permission list. You must provide either "query" or "next_token".
...
+}
+```
+
+`prisma-cloud-access-key-disable`
+
+#### Input
+
+| **Argument Name** | **Description**                                              | **Required** |
+|-------------------|--------------------------------------------------------------| --- |
+| access-key              | Access key id                                                | Required |
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:2942]:
```suggestion
| access-key              | Access key ID.                                                | Required | 
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 6]
```diff
@@ -2833,6 +2833,139 @@ Get permission list. You must provide either "query" or "next_token".
...
+#### Command example
+```!prisma-cloud-access-key-disable access-key=id```
+
+`prisma-cloud-access-key-enable`
+
+#### Input
+
+| **Argument Name** | **Description**                                              | **Required** |
+|-------------------|--------------------------------------------------------------| --- |
+| access-key              | Access key id                                                | Required |
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:2953]:
```suggestion
| access-key              | Access key ID.                                                | Required | 
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 7]
```diff
@@ -2833,6 +2833,139 @@ Get permission list. You must provide either "query" or "next_token".
...
+#### Command example
+```!prisma-cloud-access-key-enable access-key=id```
+
+`prisma-cloud-access-key-delete`
+
+#### Input
+
+| **Argument Name** | **Description**                                              | **Required** |
+|-------------------|--------------------------------------------------------------| --- |
+| access-key              | Access key id                                                | Required |
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:2964]:
```suggestion
| access-key              | Access key ID.                                                | Required | 
```

File: Packs/PrismaCloud/ReleaseNotes/4_3_9.md [hunk 8]
```diff
@@ -0,0 +1,10 @@
+
+#### Integrations
+
+##### PrismaCloud IAM
+- Added 5 commands: 
+    - 'prisma-cloud-access-key-create'
+    - 'prisma-cloud-access-keys-list'
+    - 'prisma-cloud-access-key-disable'
+    - 'prisma-cloud-access-key-enable'
+    - 'prisma-cloud-access-key-delete
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/4_3_9.md:10]:
```suggestion
Added 5 commands: 
- 'prisma-cloud-access-key-create'
- 'prisma-cloud-access-keys-list'
- 'prisma-cloud-access-key-disable'
- 'prisma-cloud-access-key-enable'
- 'prisma-cloud-access-key-delete
```

**Comment by ShirleyDenkberg** [comment]
@eyalpalo Doc review completed.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 9]
```diff
@@ -328,6 +329,54 @@ def error_file_list_request(self, repository: str, source_types: List[str], cicd
 
         return self._http_request('POST', 'code/api/v1/errors/files', json_data=data, headers=headers)
 
+    def access_key_creation(self, args: Dict[str, Any]) -> Dict[str, Any]:
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:332]: For all methods which are not straight forward, let add docstring explaining what is being done in the method (relevant arguments etc)

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 10]
```diff
@@ -1503,7 +1552,7 @@ def config_search_command(client: Client, args: Dict[str, Any]) -> CommandResult
     demisto.debug(f'Searching for config with the following params: {query=}, {limit=}, {time_filter=}, {include_resource_json=},'
                   f' {include_additional_resource_fields=}')
     response = client.config_search_request(time_filter, str(query), limit, search_id, sort_direction, sort_field,
-                                            include_resource_json,)
+                                            include_resource_json, )
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:1555]: Please revert

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 11]
```diff
@@ -1999,6 +2048,103 @@ def fetch_incidents(client: Client, last_run: Dict[str, Any], params: Dict[str,
     return incidents, ids_to_insert, updated_last_run_time
 
 
+def access_key_create_command(client: Client, args: Dict[str, Any]) -> CommandResults:
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2051]: Please add docstring for all those methods

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 12]
```diff
@@ -1999,6 +2048,103 @@ def fetch_incidents(client: Client, last_run: Dict[str, Any], params: Dict[str,
...
+    access_key = client.access_key_creation(args)
+    transform_access_keys = {
+        'Id': access_key.get('id'),
+        'Secret Key': access_key.get('secretKey'),
+    }
+
+    markdown_table = tableToMarkdown('PrismaCloud Access Key Creation', transform_access_keys, headers=['Id', 'Secret Key'])
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2058]: If the response contains only id and secretKey, let's use headerTranform of tableToMarkdown instead of transform_access_keys. look for examples in our repo.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 13]
```diff
@@ -1999,6 +2048,103 @@ def fetch_incidents(client: Client, last_run: Dict[str, Any], params: Dict[str,
...
+
+def get_access_key_by_id(client: Client, args: Dict[str, Any]) -> CommandResults:
+    access_key = client.get_access_key_by_id(args)
+    transform_access_key = {
+        'ID': access_key.get('id'),
+        'Name': access_key.get('name'),
+        'Expires On': timestamp_to_datestring(access_key.get('expiresOn'), DATE_FORMAT)
+    }
+
+    markdown_table = tableToMarkdown('PrismaCloud Access Key', transform_access_key, headers=['ID', 'Name', 'Expires On'])
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2080]: Same not regarding the markdown table. You can edit the expires on field and use the headerTransform.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 14]
```diff
@@ -1999,6 +2048,103 @@ def fetch_incidents(client: Client, last_run: Dict[str, Any], params: Dict[str,
...
+                'expiresOn') != 0 else '',
+            'Role id': access_key.get('role', {}).get('id'),
+            'Role Name': access_key.get('role', {}).get('name'),
+            'Role Type': access_key.get('roleType'),
+            'Username': access_key.get('username'),
+        }
+        for access_key in limited_access_keys_list
+    ]
+
+    markdown_table = tableToMarkdown('PrismaCloud Access Keys', transform_access_keys_list,
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2112]: Same as above

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 15]
```diff
@@ -1999,6 +2048,103 @@ def fetch_incidents(client: Client, last_run: Dict[str, Any], params: Dict[str,
...
+        outputs_key_field='id',
+        outputs=access_key,
+        raw_response=access_key,
+        readable_output=markdown_table,
+    )
+
+
+def get_access_keys_list(client: Client, args: Dict[str, Any]) -> CommandResults:
+    access_keys = client.get_access_keys_list()
+    limit = int(args.get('limit', ACCESS_KEYS_LIST_DEFAULT_LIMIT))
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2092]: Does the API support limit? or is it implemented only locally on our side? If the API supports it lets send it to the API instead getting all keys and slicing

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 16]
```diff
@@ -1999,6 +2048,103 @@ def fetch_incidents(client: Client, last_run: Dict[str, Any], params: Dict[str,
...
+            'Role Type': access_key.get('roleType'),
+            'Username': access_key.get('username'),
+        }
+        for access_key in limited_access_keys_list
+    ]
+
+    markdown_table = tableToMarkdown('PrismaCloud Access Keys', transform_access_keys_list,
+                                     headers=['ID', 'Name', 'Created By', 'Created Timestamp', 'Last Used Time', 'Status',
+                                              'Expires On', 'Role id', 'Role Name', 'Role Type', 'Username'])
+    return CommandResults(
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2115]: Please validate that in the next scenario:
1. creating a new access key
2. list all keys

There is no duplication - which means the same access key is appearing twice in the context.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 17]
```diff
@@ -2232,6 +2378,12 @@ def main() -> None:
 
             'get-remote-data': get_remote_data_command,
             'update-remote-system': update_remote_system_command,
+
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2381]: Please remove empty lines

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 18]
```diff
@@ -1,6 +1,6 @@
 sectionOrder:
-- Connect
-- Collect
+  - Connect
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:2]: Please revert all changes that are not related to your change and let me know so I can review this file.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 19]
```diff
@@ -9,105 +9,105 @@ display: Prisma Cloud v2
...
+    required: false
+    type: 8
+    section: Collect
+  - name: output_old_format
+    display: Output results of V1 commands to the context data in the old format
+    defaultvalue: 'false'
+    type: 8
+    section: Connect
+    advanced: true
+    required: false
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:110]: Lets revert this change

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2_test.py [hunk 20]
```diff
@@ -531,6 +531,137 @@ def test_permission_list_command_with_next_token(mocker, prisma_cloud_v2_client)
...
+    Given: A mock PrismaCloudV2 client.
+    When: Running access_key_create_command to create an access key.
+    Then: Ensure access key created correctly.
+    """
+    from PrismaCloudV2 import access_key_create_command
+    args = {'name': 'key-name'}
+    mock_response = {'id': 'id', 'secretKey': 'secretKey'}
+    mocker.patch.object(Client, '_http_request', return_value=mock_response)
+    command_results = access_key_create_command(prisma_cloud_v2_client, args)
+    assert command_results.raw_response == mock_response
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2_test.py:545]: it is better to assert on outputs

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 21]
```diff
@@ -2833,28 +2833,172 @@ Get permission list. You must provide either "query" or "next_token".
...
+
+`prisma-cloud-access-key-disable`
+
+#### Input
+
+| **Argument Name** | **Description**                                              | **Required** |
+|-------------------|--------------------------------------------------------------| --- |
+| access-key              | Access key ID.                                                | Required | 
+
+#### Command example
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:2955]: lets add readable output example for those 3 commands

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 22]
```diff
@@ -2833,28 +2833,172 @@ Get permission list. You must provide either "query" or "next_token".
...
+```!prisma-cloud-access-key-delete access-key=id```
+
 ## Breaking changes from the previous version of this integration - Prisma Cloud v2
 The following sections list the changes in this version.
 
 ### Commands
 #### The following commands were deprecated in this version because they are not supported by the API anymore:
 * ***redlock-list-scans***
-* *redlock-get-scan-status***
+* ***redlock-get-scan-status***
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:2986]: Please revert all changes that are not should be part of you PR

File: Packs/PrismaCloud/TestPlaybooks/Prisma_Cloud_Test.yml [hunk 23]
```diff
@@ -1,19 +1,20 @@
 id: Prisma Cloud V2 Test
-version: -1
+version: 14
```
Comment (eyalpalo) [inline Packs/PrismaCloud/TestPlaybooks/Prisma_Cloud_Test.yml:2]: Lets revert all changes that are not part of your PR, run format on this file

File: Packs/PrismaCloud/TestPlaybooks/Prisma_Cloud_Test.yml [hunk 24]
```diff
@@ -1531,19 +1553,296 @@ tasks:
...
+      script: '|||prisma-cloud-access-key-enable'
+      type: regular
+      iscommand: true
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "53"
+    scriptarguments:
+      access-key:
+        simple: ${PrismaCloud.AccessKeys.id}
```
Comment (eyalpalo) [inline Packs/PrismaCloud/TestPlaybooks/Prisma_Cloud_Test.yml:1753]: which access key will it take? if we have more than 1?

**Review by eyalpalo — CHANGES_REQUESTED** [review]
Nice job! please see my comments

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 25]
```diff
@@ -55,6 +55,7 @@
 PAGE_SIZE_MAX_VALUE = 10000
 
 DEFAULT_LIMIT = '50'
+ACCESS_KEYS_LIST_DEFAULT_LIMIT = '50'
```
[the remark this answers is not in the document]
Comment (ilaredo) [reply Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:58]: to avoid affecting my code If someday someone would change the default_limit

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py (hunk 15 above)
Comment (ilaredo) [reply Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:2092]: Unfortunately API does not support limit

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml (hunk 18 above)
Comment (ilaredo) [reply Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml]: reverted

File: Packs/PrismaCloud/TestPlaybooks/Prisma_Cloud_Test.yml (hunk 23 above)
Comment (ilaredo) [reply Packs/PrismaCloud/TestPlaybooks/Prisma_Cloud_Test.yml:2]: I added tasks for testing access-key flow. it changed ID's and boxes position to the other tasks that were there.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py (hunk 25 above)
[the remark this answers is not in the document]
Comment (eyalpalo) [reply Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:58]: In most cases, we have a default limit for all list calls. If they both have a default of 50, let's stick to the default limit.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 26]
```diff
@@ -2131,9 +2131,90 @@ script:
...
     execution: false
     name: update-remote-system
+  - arguments:
+      - description: Access key name.
+        name: name
+        required: true
+      - description: Timestamp in milliseconds when access key expires. Default:0.
+        name: expires-on
+    deprecated: false
+    description: Adds a new access key for the current user. If you have API access, you can create up to two access keys.
```
Comment (eyalpalo) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:2141]: Is it correct for all kinds of users? Also for admins? if yes ignore this comment and resolve the conversation.

File: Packs/PrismaCloud/TestPlaybooks/Prisma_Cloud_Test.yml [hunk 27]
```diff
@@ -1,19 +1,20 @@
 id: Prisma Cloud V2 Test
-version: -1
+version: 15
```
Comment (eyalpalo) [inline Packs/PrismaCloud/TestPlaybooks/Prisma_Cloud_Test.yml:2]: Please revert all changes that are not yours (you can compare to master in pycharm)

**Review by eyalpalo — CHANGES_REQUESTED** [review]
Very nice!
regarding my notes - if not relevant resolve them. Please see my note regarding the test playbook

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 28]
```diff
@@ -2131,9 +2131,90 @@ script:
...
+    deprecated: false
+    description: Returns all access keys for your tenant if you have a Prisma Cloud System Admin role. Returns just your access keys if you don't have this role.
+    execution: false
+    name: prisma-cloud-access-keys-list
+    outputs:
+      - contextPath: PrismaCloud.AccessKeys.id
+        description: Access key ID.
+        type: String
+      - contextPath: PrismaCloud.AccessKeys.createdBy
+        description: Created By.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:2165]:
```suggestion
        description: The user who created the access key.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 29]
```diff
@@ -2131,9 +2131,90 @@ script:
...
+        type: String
+      - contextPath: PrismaCloud.AccessKeys.createdTs
+        description: Time access key was created.
+        type: Number
+      - contextPath: PrismaCloud.AccessKeys.expiresOn
+        description: Access key expires on.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:2171]:
```suggestion
        description: The time the access key expires.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 30]
```diff
@@ -2131,9 +2131,90 @@ script:
...
+        type: Number
+      - contextPath: PrismaCloud.AccessKeys.lastUsedTime
+        description: Access key last used.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:2174]:
```suggestion
        description: The last time the access key was used.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 31]
```diff
@@ -2131,9 +2131,90 @@ script:
...
+        type: Number
+      - contextPath: PrismaCloud.AccessKeys.name
+        description: Name.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:2177]:
```suggestion
        description: The name of the access key.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml [hunk 32]
```diff
@@ -2131,9 +2131,90 @@ script:
...
+        type: String
+      - contextPath: PrismaCloud.AccessKeys.role
+        description: User role.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.yml:2180]:
```suggestion
        description: Access key user role.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 33]
```diff
@@ -2833,6 +2833,162 @@ Get permission list. You must provide either "query" or "next_token".
 >### Next Page Token:
 >token2
 
+### Access Keys
+
+***
+Access Keys are a secure way to enable programmatic access to the Prisma Cloud API. By default, only the System Admin has API access and can enable API access for other administrators. If you have API access, you can create up to two access keys.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:2839]:
```suggestion
Access keys are a secure way to enable programmatic access to the Prisma Cloud API. By default, only the System Admin has API access and can enable API access for other administrators. If you have API access, you can create up to two access keys.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 34]
```diff
@@ -2833,6 +2833,162 @@ Get permission list. You must provide either "query" or "next_token".
...
+
+#### Context Output
+
+| **Path**                           | **Type** | **Description**             |
+|------------------------------------|----------|-----------------------------|
+| PrismaCloud.AccessKeys.id          | String   | Access key ID.               | 
+| PrismaCloud.AccessKeys.name        | String   | The name of the access key.                        |
+| PrismaCloud.AccessKeys.createdBy   | String   | The user who created the access key.                  | 
+| PrismaCloud.AccessKeys.createdTs   | String   | Time access key was created. |
+| PrismaCloud.AccessKeys.expiresOn   | String   | Access key expires on.       |
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:2897]:
```suggestion
| PrismaCloud.AccessKeys.expiresOn   | String   | The time the access key expires.       |
```

**Comment by ShirleyDenkberg** [comment]
@ilaredo @eyalpalo  Doc re-reviewed as requested.

**Review by eyalpalo — APPROVED** [review]
Very nice!
